posts/create-a-modern-web-app-using-vuejs-and-spring-boot-with-jhipster.md

```diff
@@ -0,0 +1,172 @@
+---
+authors:
+- Theo Lebrun
+tags:
+- Vue.js
+- Spring Boot
+- JHipster
+date: 2019-02-14T12:12:12.000Z
+title: "Create a modern Web app using Vue.js and Spring Boot with JHipster"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2019/02/vuejs-jhipster-logo.png
+---
+
+[Vue.js](https://vuejs.org/) is the new trending framework in the Javascript front-end world and its simplicity lets you build web applications pretty fast. The structure of a Vue.js application is very similar to an Angular application as it uses components and templates to build pages. [This video](https://www.vuemastery.com/courses/intro-to-vue-js/vue-instance) is a good introduction to start playing with Vue.js especially if you never used Angular or React before.
+
+Here is a size comparison (minified) of the current popular frameworks ([more details here](https://gist.github.com/Restuta/cda69e50a853aa64912d)):
+
+```text
+| Name                     | Size  |
+| ------------------------ | ----- |
+| Angular 2                | 566K  |
+| Ember 2.2.0              | 435K  |
+| React 0.14.5 + React DOM | 133K  |
+| React 16.2.0 + React DOM |  97K  |
+| Vue 2.4.2                |  58K  |
+| Inferno 1.2.2            |  48K  |
+| Preact 7.2.0             |  16K  |
+```
+
+As I said before, the simplicity of Vue.js makes it very lightweight and easy to integrate with other libraries or existing projects. On the other hand, Vue.js is totally capable of building robust Single-Page Applications with the right tools and [JHipster](https://www.jhipster.tech/) is the perfect match for that. If you're not familiar with JHipster, I recommend watching [this video](https://www.youtube.com/watch?v=-VQ_SVkaXbs) that will give you a good introduction on what you can do with JHipster.
+
+# Vue.js using a JHipster blueprint
+
+By default, JHipster asks you to choose between Angular or React for the front-end framework. The concept of blueprint was introduced in version 5 and the goal is to extend the functionality of JHipster by letting you use your own sub-generator. More information on how to create and use a blueprint can be found in [the documentation](https://www.jhipster.tech/modules/creating-a-blueprint/).
+
+## Installation
+
+The official Vue.js blueprint will be used to generate the front-end. You can follow the installation instruction on [the blueprint repository](https://github.com/jhipster/jhipster-vuejs). Since the blueprint was still in development when I wrote this blog post, you may encounter some issues. Feel free to report them and submit a Pull Request if you think that a part was done the wrong way.
```

Comment:
```suggestion
The official Vue.js blueprint is used to generate the front-end. You can follow the installation instruction on [the blueprint repository](https://github.com/jhipster/jhipster-vuejs). Since the blueprint was still in development when I wrote this blog post, you may encounter some issues. Feel free to report them and submit a Pull Request if you think that a part was done the wrong way.
```